### ISSUE TITLE
Publish added vector layers as WFS in QGIS Server

### ISSUE THREAD

**Issue by JakobMiksch** [pblottiere/QSA#64]
Adding vector layers to a QGIS project file works as expected. But they are not published as WFS in the QGIS Server settings. See screenshot below.

Is it possible to use the QSA API to publish a vector layer as WFS in the QGIS Server settings block?

![image](https://github.com/user-attachments/assets/ec37fe31-617a-46d7-be46-a006d8a99fe4)

**Comment by JakobMiksch** [comment]
@pblottiere I am considering to implement this.

My idea would be to add a `wfs` property to the "layer definition" so it would look like this:

- `type`
- `name`
- `datasource`
- `overview` (optional)
- `crs` (optional)
- `publication`
  - `wfs`
    - `published` (boolean)
    - `precision` (integer)
    - in future: transaction (update, insert, delete)
  - in future: `wms`
  - in future: `wmts`
  - in future: `wcs`

what do you think about that?

**Comment by pblottiere** [comment]
Hello @JakobMiksch,

> I am considering to implement this.

Nice :tada: 

> what do you think about that?

Yes indeed it's a good idea :+1:. I'm just afraid to clutter the layer definition :thinking:.

Another way would be to add a specific service endpoint for layers. For example a POST `/api/projects/<project_name>/layers/<layer_name>/wfs` with a payload for options and configuration. 

```
"published": True,
"geometry_precision": 8,
"update": True,
...
```

For WMTS with `/api/projects/<project_name>/layers/<layer_name>/wmts`:

```
"published": True,
"PNG": True,
"JPEG": False
```

 And by default a layer could only be published in WMS? What do you think?

**Comment by JakobMiksch** [comment]
Your approach looks good as well. 

QGIS published WMS by default, that's why I would leave this behavior with QSA as well. However, it should be possible to deactivate the publication of WMS for a layer.

Another question:
how would should a user find out if a layer is for example published via WFS?

would that be in the response of `GET /api/projects/{project}/layers/{layer}` ?

**Comment by pblottiere** [comment]
> QGIS published WMS by default, that's why I would leave this behavior with QSA as well. However, it should be possible to deactivate the publication of WMS for a layer.

Yes, I agree :+1:.  A POST `/api/projects/<project_name>/layers/<layer_name>/wms` entrypoint should be made available too.

> how should a user find out if a layer is for example published via WFS?

To be consistent with the "publish" API, a GET `/api/projects/{project}/layers/{layer}/wfs` entrypoint would probably be enough.
